### PR TITLE
Airgap support

### PR DESCRIPTION
**Pull request by emosbaugh**

### CODE REVIEW

File: cmd/kots/cli/ingress.go [hunk 1]
```diff
@@ -83,11 +83,11 @@ func IngressInstallCmd() *cobra.Command {
...
 
 			if identityConfig.Enabled {
-				log.ChildActionWithSpinner("Deploying the Identity Service")
+				log.ChildActionWithSpinner("Configuring the Identity Service")
 
-				// we have to re-deploy the identity service if kotsadm ingress is changing
-				if err := identity.Deploy(cmd.Context(), log, clientset, namespace, *identityConfig, ingressConfig); err != nil {
-					return errors.Wrap(err, "failed to deploy identity service")
+				// we have to update the dex secret if kotsadm ingress is changing because it relies on the redirect uri
+				if err := identity.DeploySecret(cmd.Context(), log, clientset, namespace, *identityConfig, ingressConfig); err != nil {
```
Comment (salahalsaleh) [inline cmd/kots/cli/ingress.go:89]: this function will no longer exist, it was reverted to the "apply" method instead

File: cmd/kots/cli/identity.go [hunk 2]
```diff
@@ -95,6 +101,81 @@ func IdentityServiceInstallCmd() *cobra.Command {
...
+
+			identityConfig.Enabled = true
+			identityConfig.DisablePasswordAuth = true
+			identityConfig.IngressConfig.Enabled = true
+
+			if err := identity.SetConfig(cmd.Context(), namespace, identityConfig); err != nil {
+				return errors.Wrap(err, "failed to set identity config")
+			}
+
+			if err := identity.DeploySecret(cmd.Context(), log, clientset, namespace, identityConfig, *ingressConfig); err != nil {
```
Comment (salahalsaleh) [inline cmd/kots/cli/identity.go:167]: same here

Comment (emosbaugh) [reply cmd/kots/cli/identity.go:167]: why? that means you will have to include all registry settings for airgap each time you want to make a change to the identity config. is there some reason you need to remove that function?

Comment (emosbaugh) [reply cmd/kots/cli/identity.go]: i added the function back

**Review by salahalsaleh — APPROVED** [review]

**Pull request merged**